xml/System.Web.Script.Serialization/JavaScriptTypeResolver.xml

```diff
@@ -31,14 +31,236 @@
+                return type.FullName;
+            }
+
+            throw new InvalidOperationException("Cannot serialize an object of type " + type + ". Did you forget to add it to the allow list?");
+        }
+    }
+}
+```
+
+The preceding app outputs the following to the console, formatted for readability.
+
+```txt
+Serialized:
+[
+    {
+        "__type": "SampleApp.Person",
+        "Name": "Kristen Solstad",
+        "Age": 15,
+        "HomeAddress": {
+            "__type": "SampleApp.Address",
+            "Street1": "123 Palm Ave",
+            "Street2": null,
+            "City": "Some City",
+            "StateOrProvince": "ST",
+            "Country": "United States",
+            "PostalCode": "00000"
+        }
+    },
+    {
+        "__type": "SampleApp.Adult",
+        "Occupation": "Mechanic",
+        "Name": "Alex Johnson",
+        "Age": 39,
+        "HomeAddress": {
+            "__type": "SampleApp.Address",
+            "Street1": "445 Lorry Way",
+            "Street2": "Unit 3A",
+            "City": "Some City",
+            "StateOrProvince": null,
+            "Country": "United Kingdom",
+            "PostalCode": "AA0 A00"
+        }
+    }
+]
+
+Deserialized 2 people.
+Kristen Solstad (Age 15) [SampleApp.Person]
+Alex Johnson (Age 39) [SampleApp.Adult]
+```
+
+In the preceding sample, the `Adult` type subclasses the `Person` type. A custom `JavaScriptTypeResolver` is used to include the type information as part of the generated JSON payload. This allows limited polymorphism when deserializing the JSON payload back into a .NET object graph. The payload can control whether to return a base `Person` instance or a derived `Adult` instance back to the caller.
+
+This sample is safe because it uses an `allow-list` mechanism to control deserialization. The code:
+
+* Initializes the `CustomTypeResolver` with an explicit list of allowed types.
+* Restricts the deserialization process to only approved list of types. The restriction prevents [deserialization attacks](https://owasp.org/www-community/vulnerabilities/Deserialization_of_untrusted_data), where the remote client specifies a malicious `__type` in the JSON payload and tricks the server into deserializing a dangerous type.
+
+Even though the app only expects `Person` and `Adult` instances to be deserialized as part of the top-level array, it's still necessary to add `Address` to the allow-list because:
+
+* Serializing a `Person` or `Adult` also serializes an `Address` as part of the object graph.
+* All types that might be present in the object graph need to be accounted for in the allow list. Primitives like `int` and `string` do not need to be specified.
+
+> [!WARNING]
+>  Do not call `Type.GetType(id)` within the `ResolveType` method. This could introduce a security vunerability into the app. Instead, iterate through the list of allowed types, comparing their `Type.FullName` property against the incoming `id`, as shown in the preceding sample:
```

Comment:
```suggestion
>  Do not call `Type.GetType(id)` within the `ResolveType` method. This could introduce a security vunerability into the app. Instead, iterate through the list of allowed types and compare their `Type.FullName` property against the incoming `id`, as shown in the preceding sample.
```